bigcommerce/resources/base.py

```diff
@@ -124,4 +124,11 @@ def delete(self):
     @classmethod
     def delete_all(cls, parentid, connection=None):
         path = "%s/%s/%s" % (cls.parent_resource, parentid, cls.resource_name)
-        return cls._make_request('DELETE', path, connection)
\ No newline at end of file
+        return cls._make_request('DELETE', path, connection)
+
+class CountableApiResource(ApiSubResource):
```

Comment:
There are others as well, such as blog posts, which we apparently don't have a resource for in this library. Here is the full list of resources and subresources with a `/count` endpoint, according to [the documentation](https://developer.bigcommerce.com/api)

* BlogPost
* Redirect
* Customer
* CustomerGroup
* Country
* State
* Coupon
* Order
* OrderProduct
* Shipment
* Brand
* BulkPricingRule
* Category
* ConfigurableField
* CustomField
* Option
* OptionSet
* Product
* ProductImage
* ProductRule
* ProductVideo
* ProductSKU

Unfortunately, not all of these are available in this library. However, I think it makes sense to add `CountableApiResource` to those that do exist. To do so, we'll probably need to create a `CountableApiSubResource` as well.